components/admin/users/AdminUsersTable.tsx

```diff
@@ -4,132 +4,167 @@ import _ from 'lodash'
   isAdmin,
   setUsers,
-  index,
+  usersIndex,
   users,
   id
 }) => {
   const [changeRights] = useMutation(changeAdminRights)
 
+  const newAdminRights = isAdmin ? 'false' : 'true'
+
   const mutationVariable = {
     variables: {
       id: parseInt(id + ''),
-      status: isAdmin ? 'false' : 'true'
+      status: newAdminRights
     }
   }
 
   const changeButton = async () => {
     await changeRights(mutationVariable)
-    const newUsers = users && [...users]
-    if (newUsers) {
-      newUsers[index].isAdmin = isAdmin ? 'false' : 'true'
-    }
+    const newUsers = [...users]
+    newUsers[usersIndex].isAdmin = newAdminRights
     setUsers(newUsers)
   }
 
   return (
     <Button type={isAdmin ? 'danger' : 'success'} onClick={changeButton}>
-      {(isAdmin ? 'Remove' : 'Grant') + ' Admin Rights'}
+      {isAdmin ? 'Remove' : 'Add'}
     </Button>
   )
 }
 
-const RowData: React.FC<RowDataProps> = ({ user, users, setUsers, index }) => {
+const RowData: React.FC<RowDataProps> = ({
+  user,
+  users,
+  setUsers,
+  usersIndex,
+  searchTerm,
+  option
+}) => {
+  option = _.lowerCase(option)
+
   const data = userProperties.map((property: string, key: number) => {
     let value = user[property]
 
-    const displayOption =
-      property !== 'isAdmin' ? (
-        value
-      ) : (
+    if (searchTerm && property === option) {
+      value = AdminUsersSplitSearch(value, searchTerm)
+    }
+
+    if (property === 'isAdmin')
+      value = (
         <AdminOption
           isAdmin={user[property] === 'true'}
           setUsers={setUsers}
-          index={index}
+          usersIndex={usersIndex}
           users={users}
-          id={users && users[index].id}
+          id={users[usersIndex].id}
         />
       )
 
     return (
-      <td style={{ verticalAlign: 'middle' }} key={key}>
-        {displayOption}
+      <td className="align-middle" key={key}>
+        {value}
       </td>
     )
   })
 
   return <>{data}</>
 }
 
-const UsersList: React.FC<UsersListProps> = ({ users, setUsers }) => {
-  const list =
-    users &&
-    users.reduce((acc: any[], user: any, usersIndex: number) => {
-      acc.push(
-        <tr key={usersIndex} style={{ textAlign: 'center' }}>
-          <RowData
-            user={user}
-            setUsers={setUsers}
-            index={usersIndex}
-            users={users}
-          />
-        </tr>
-      )
+const UsersList: React.FC<UsersListProps> = ({
+  users,
+  setUsers,
+  searchOption
+}) => {
+  const { searchTerm, admin } = searchOption
+  let { option } = searchOption
+  option = _.lowerCase(option)
+
+  const list = users.reduce((acc: any[], user: any, usersIndex: number) => {
+    if (searchTerm) {
+      const value = user[option]
+      if (!value.includes(searchTerm)) return acc
+    }
+
+    const { isAdmin } = user
+
+    if (admin === 'Non-Admins' && isAdmin === 'true') return acc
+    if (admin === 'Admins' && isAdmin === 'false') return acc
```

Comment:
I don't think you need an `isAdmin` check here because non-admins won't even be able to see the page.